R/Angel Chen/ticket #20492 Norman.R

```diff
@@ -0,0 +1,144 @@
+
+doc$dataset$project <- eml_nsf_to_project("1523059")
+
+eml_validate(doc)
+doc_path <- file.path(tempdir(), 'science_metadata.xml')
+write_eml(doc, doc_path)
+
+update <- publish_update(adc, 
+                         metadata_pid = pkg$metadata,
+                         resource_map_pid = pkg$resource_map,
+                         data_pids = pkg$data,
+                         metadata_path = doc_path, 
+                         public = FALSE)
+#https://arcticdata.io/catalog/view/urn%3Auuid%3Afff9acca-9e4c-481a-8186-678892ca5a6e
+
+pkg <- get_package(adc, 'resource_map_urn:uuid:fff9acca-9e4c-481a-8186-678892ca5a6e', file_names = TRUE)
+doc <- read_eml(getObject(adc, pkg$metadata))
+emld::eml_version("eml-2.1.1")
+
+
+doc$dataset$coverage$geographicCoverage[[2]]$boundingCoordinates$westBoundingCoordinate <- "-156.51550"
+doc$dataset$coverage$geographicCoverage[[2]]$boundingCoordinates$eastBoundingCoordinate <- "-156.64143"
+
+eml_validate(doc)
+doc_path <- file.path(tempdir(), 'science_metadata.xml')
+write_eml(doc, doc_path)
+
+update <- publish_update(adc, 
+                         metadata_pid = pkg$metadata,
+                         resource_map_pid = pkg$resource_map,
+                         data_pids = pkg$data,
+                         metadata_path = doc_path, 
+                         public = FALSE)
+#https://arcticdata.io/catalog/view/urn%3Auuid%3A8b86d811-2b0e-48ae-9f48-7a57d67eccfe
+
+pkg <- get_package(adc, 'resource_map_urn:uuid:8b86d811-2b0e-48ae-9f48-7a57d67eccfe', file_names = TRUE)
+doc <- read_eml(getObject(adc, pkg$metadata))
+emld::eml_version("eml-2.1.1")
+
+doc$dataset$coverage$geographicCoverage[[2]]$boundingCoordinates$northBoundingCoordinate <- "69.057876"
+doc$dataset$coverage$geographicCoverage[[2]]$boundingCoordinates$southBoundingCoordinate <- "69.057876"
+doc$dataset$coverage$geographicCoverage[[2]]$boundingCoordinates$westBoundingCoordinate <- "-152.862827"
+doc$dataset$coverage$geographicCoverage[[2]]$boundingCoordinates$eastBoundingCoordinate <- "-152.862827"
+
+eml_validate(doc)
+doc_path <- file.path(tempdir(), 'science_metadata.xml')
+write_eml(doc, doc_path)
+
+update <- publish_update(adc, 
+                         metadata_pid = pkg$metadata,
+                         resource_map_pid = pkg$resource_map,
+                         data_pids = pkg$data,
+                         metadata_path = doc_path, 
+                         public = FALSE)
+#https://arcticdata.io/catalog/view/urn:uuid:5b6546c3-b8a6-420f-b1ba-1422c4479f6b
+
+#https://arcticdata.io/catalog/view/urn%3Auuid%3Adda6d1ed-b817-42ad-9c23-a33b01c59a88
+
+pkg <- get_package(adc, 'resource_map_urn:uuid:160fbcbf-92af-4d3b-8ebf-6092a4435a27', file_names = TRUE)
+doc <- read_eml(getObject(adc, pkg$metadata))
+emld::eml_version("eml-2.1.1")
+
+set_rights_and_access(adc, pids = unlist(pkg), subject = 'http://orcid.org/0000-0002-5718-6032')
```

Comment:
does just writing the subject with http:// instead of https:// replace the need for this step: `subject <- sub("^https://", "http://", subject)`?

Reply:
yes it is convenient for if you have lots of orcids